### PR TITLE
[chronoengine] New port

### PR DESCRIPTION
**Pull request by talregev**
**Describe the pull request**
Add chrono port.

- #### What does your PR fix?
  Fixes #...

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  <all / linux, windows, ...>, <Yes/No>

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  `Your answer`

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  **Yes**

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

**Comment by talregev** [comment]
I am thinking to cancel this PR. this library, is very demanding on the compilation. Long time with a lot of memory.

**Comment by talregev** [comment]
@FrankXie05

**Review by vicroms — DISMISSED** [review]

**Review by BillyONeal — CHANGES_REQUESTED** [review]
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)
- [ ] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.  
  
The name 'chrono' is already taken by a time tracking tool. Would you accept one of the following names?
    * `chrono-physics-simulation-engine` https://repology.org/project/chrono-physics-simulation-engine/
    * `chronoengine` https://repology.org/project/chronoengine/
  (I submitted a report to repology that these entries should be merged)  
  
- [ ] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)  
  
```
src\chrono_mumps\building_Mumps\CMakeLists_5.0.2.txt:81:find_package(BLAS)
src\chrono_mumps\building_Mumps\CMakeLists_5.0.2.txt:104:find_package(OpenMP)
src\chrono_mumps\building_Mumps\CMakeLists_5.1.1.txt:81:find_package(BLAS)
src\chrono_mumps\building_Mumps\CMakeLists_5.1.1.txt:124:find_package(OpenMP)
src\chrono_mumps\CMakeLists.txt:91:find_package(BLAS)
src\chrono_opengl\CMakeLists.txt:44:find_package(OpenGL)
src\chrono_opengl\CMakeLists.txt:45:find_package(GLM)
src\chrono_opengl\CMakeLists.txt:46:find_package(GLEW)
src\chrono_opengl\CMakeLists.txt:47:find_package(GLFW)
src\chrono_pardisomkl\CMakeLists.txt:30:find_package(MKL)
src\chrono_pyparser\CMakeLists.txt:41:find_package(PythonLibs ${PYTHON_VERSION_STRING} EXACT)
src\chrono_sensor\CMakeLists.txt:32:find_package(GLFW OPTIONAL_COMPONENTS)
src\chrono_sensor\CMakeLists.txt:33:find_package(OpenGL OPTIONAL_COMPONENTS)
src\chrono_sensor\CMakeLists.txt:34:find_package(GLEW OPTIONAL_COMPONENTS)
src\chrono_swig\chrono_python\CMakeLists.txt:76:find_package(PythonLibs ${PYTHON_VERSION_STRING} EXACT)
src\CMakeLists.txt:99:find_package(Threads)
src\CMakeLists.txt:115:find_package(OpenMP)
src\CMakeLists.txt:166: find_package("TBB")
src\CMakeLists.txt:180:   find_package(SIMD)
src\CMakeLists.txt:253:find_package(Eigen3 3.3.0)
src\CMakeLists.txt:296:find_package(MPI)
src\CMakeLists.txt:316:find_package(CUDA QUIET)
src\CMakeLists.txt:386:  find_package(Thrust)
src\CMakeLists.txt:503:    find_package(HDF5 COMPONENTS CXX)
src\CMakeLists.txt:510:        find_package(HDF5 NAMES hdf5 COMPONENTS CXX ${SEARCH_TYPE})
```
  These are all very concerning. Can you explain why these optional dependencies are not enumerated in `vcpkg.json`? To clarify: `vcpkg install blas && vcpkg install chrono` and `vcpkg install chrono && vcpkg install blas` need to produce the same output, which is broken when a build system engages different behavior silently based on what's already provided.  
  
- [ ] The versioning scheme in `vcpkg.json` matches what upstream says.  
  
The latest version known to upstream is `7.0.3`, why is this declaring itself `8.0.0`?
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See docs/examples/adding-an-explicit-usage.md for context.  
  
No usage. You might want to add some.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

**Comment by talregev** [comment]
@BillyONeal 
There many modules that are optional. Some of them required matlab or cuda that I don't have to check.
One port is depends on develop branch of that port witch vcpkg don't have.
So I made this port with 2 features. If in the future people need to have more modules, they can add more features to this port.


I see the release version is 7.0.3, But they release another version after that with 8.0.0.
I know when I add a port you want the most release.
So I assume that 8.0.0 is more new.
I can go back to 7.0.3 and we can release 8.0.0 or other versions later.

**Comment by talregev** [comment]
I will change it to chrono engine.
I will change to 7.0.3. we can always update later.

**Comment by talregev** [comment]
@BillyONeal 
Chrono just release the `8.0.0` version. So I switch back to 8.

**Comment by BillyONeal** [comment]
>There many modules that are optional. Some of them required matlab or cuda that I don't have to check.

OK then we should patch the build system or set `CMAKE_DISABLE_FIND_PACKAGE_Xxx` or similar to ensure those detections always say "false" so that people don't get a different thing installed if they happen to have matlab or cuda present on their system. We don't care that every optional knob is available when something is added, just that people don't randomly get extra knobs we haven't tested turned on just because they happened to build on a different machine.

> @BillyONeal Chrono just release the `8.0.0` version. So I switch back to 8.

Excellent, thank you!

**Comment by talregev** [comment]
> > There many modules that are optional. Some of them required matlab or cuda that I don't have to check.
> 
> OK then we should patch the build system or set `CMAKE_DISABLE_FIND_PACKAGE_Xxx` or similar to ensure those detections always say "false" so that people don't get a different thing installed if they happen to have matlab or cuda present on their system. We don't care that every optional knob is available when something is added, just that people don't randomly get extra knobs we haven't tested turned on just because they happened to build on a different machine.
> 
> > @BillyONeal Chrono just release the `8.0.0` version. So I switch back to 8.
> 
> Excellent, thank you!

As far I understand, the optional detection is only activate if I enable the modules. All the module is off by default. 
You can take a look in their module documentations: 
https://api.projectchrono.org/development/tutorial_table_of_content_install.html

File: ports/chronoengin/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "chronoengin",
```
Comment (BillyONeal) [inline ports/chronoengin/vcpkg.json:2]:
```suggestion
  "name": "chronoengine",
```

Comment (talregev) [reply ports/chronoengin/vcpkg.json]: Thanks! I also fix the port name.

**Comment by talregev** [comment]
> > > There many modules that are optional. Some of them required matlab or cuda that I don't have to check.
> > 
> > 
> > OK then we should patch the build system or set `CMAKE_DISABLE_FIND_PACKAGE_Xxx` or similar to ensure those detections always say "false" so that people don't get a different thing installed if they happen to have matlab or cuda present on their system. We don't care that every optional knob is available when something is added, just that people don't randomly get extra knobs we haven't tested turned on just because they happened to build on a different machine.
> > > @BillyONeal Chrono just release the `8.0.0` version. So I switch back to 8.
> > 
> > 
> > Excellent, thank you!
> 
> As far I understand, the optional detection is only activate if I enable the modules. All the module is off by default. You can take a look in their module documentations: https://api.projectchrono.org/development/tutorial_table_of_content_install.html

@BillyONeal Let me know if you agree with me.

**Comment by BillyONeal** [comment]
> > As far I understand, the optional detection is only activate if I enable the modules. All the module is off by default. You can take a look in their module documentations: https://api.projectchrono.org/development/tutorial_table_of_content_install.html
> 
> @BillyONeal Let me know if you agree with me.

There are many many unguarded `find_package` calls. Even just looking at `8.0.0-6dccb3d031.clean\src\CMakeLists.txt` has:

```
find_package("TBB")
find_package(SIMD) # Guarded by an on by default option USE_SIMD, there is a port for this
find_package(Eigen3 3.3.0) # OK, an explicit dependency
find_package(MPI)
find_package(CUDA QUIET)
find_package(Thrust)
find_package(HDF5 COMPONENTS CXX) # OK, guarded by off-by-default option ENABLE_HDF5
```

This stuff needs to not build a subtly different package because someone happens to have some of these installed first.

**Comment by BillyONeal** [comment]
(To clarify, you don't need to mess with every optional dependency that is indeed guarded by an off option. You do, however, need to demonstrate that those are so guarded. Patching more things to be `REQUIRED` is always going to be appreciated even if they happen to be not touched by that particular build, because it clearly demonstrates that something is, in fact, disabled)

**Comment by talregev** [comment]
> (To clarify, you don't need to mess with every optional dependency that is indeed guarded by an off option. You do, however, need to demonstrate that those are so guarded. Patching more things to be `REQUIRED` is always going to be appreciated even if they happen to be not touched by that particular build, because it clearly demonstrates that something is, in fact, disabled)

I will create a patch.

**Comment by talregev** [comment]
@BillyONeal Please review.

**Review by BillyONeal — APPROVED** [review]
```
.\vcpkg.exe install chronoengine:x64-windows pass
.\vcpkg.exe install chronoengine[irrlicht]:x64-windows pass
.\vcpkg.exe install chronoengine[vehicle]:x64-windows pass
.\vcpkg.exe install chronoengine[irrlicht,vehicle]:x64-windows pass
```

Thanks for the new port!

**Pull request merged**